prov/rxm/src/rxm_init.c

```diff
@@ -187,6 +194,16 @@ int rxm_info_to_rxm(uint32_t version, const struct fi_info *core_info,
 			return -FI_ENOMEM;
 	}
 
+	if (core_info->caps & FI_HMEM) {
+		info->caps &= ~FI_COLLECTIVE;
+		info->tx_attr->caps &= ~FI_COLLECTIVE;
+		info->rx_attr->caps &= ~FI_COLLECTIVE;
+	} else {
+		info->caps &= ~FI_HMEM;
+		info->tx_attr->caps &= ~FI_HMEM;
+		info->rx_attr->caps &= ~FI_HMEM;
+	}
```

Comment:
This is not needed.  FI_COLLECTIVE has its own fi_info structure reported by rxm.  If FI_HMEM cannot be combined with FI_COLLECTIVE, then the place to remove FI_COLLECTIVE is where those fi_info caps are defined.  This ensures that the core properly handles requests from the app where FI_COLLECTIVE | FI_HMEM are requested.

For example, see rxm_tx_attr_coll, rxm_rx_attr_coll, etc.